src/HaroohiePals.MarioKartToolbox/Gui/View/PropertyGrid/MapObjectIdEditor.cs

```diff
@@ -59,8 +59,9 @@ public bool Draw(string label, PropertyEditorContext context)
 
         if (_selectorWindow.Confirm)
         {
-            context.TempActions.Add(
-                new SetPropertyAction(context.Property.SourceObjects, context.Property.Name, _selectorWindow.MapObjectId));
+            bool TTVisible = _mobjDatabase.GetById(_selectorWindow.MapObjectId).IsTimeTrialVisible;
+            context.TempActions.Add(new SetPropertyAction(context.Property.SourceObjects, context.Property.Name, _selectorWindow.MapObjectId));
+            context.TempActions.Add(new SetPropertyAction(context.Property.SourceObjects, "TTVisible", TTVisible));
```

Comment:
Leaving the property name as a string will cause issues in case TTVisible name changes, reflecting changes won't be done automatically. you can use ``nameof(MkdsMapObject.TTVisible)``.